### PR TITLE
STYLE: Replace (*p).member with p->member

### PR DESCRIPTION
**Pull request by hjmjohnson**
Simplified code by using the arrow operator, instead of using
parentheses, asterisk and dot operators.

Did so by replacing the regular expression `\(\*(\w+)\)\.` with `->`,
followed by a few manual fixes.

```
find . \( -iname "*.cxx" -or -iname "*.hxx" -or -iname "*.h" \) -exec perl -pi -w -e 's/\(\*(\w+)\)\./->/g' {} \;
```

### CODE REVIEW

**Review by pieper — APPROVED** [review]
Definitely looks cleaner. 👍

File: Libs/MRML/Widgets/qMRMLSortFilterProxyModel.cxx [hunk 1]
```diff
@@ -272,7 +272,7 @@ ::filterAcceptsNode(vtkMRMLNode* node)const
     if (!node->IsA(nodeType.toUtf8().data()))
       {
       //std::cout << "Reject node: " << node->GetName() << "(" << node->GetID()
-      //          << ") type: " << typeid(*node).name() <<std::endl;
+      //          << ") type: " << typeidnode->name() <<std::endl;
```
Comment (jcfr) [inline Libs/MRML/Widgets/qMRMLSortFilterProxyModel.cxx:275]: This change should likely be reverted

File: Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx [hunk 2]
```diff
@@ -299,15 +299,15 @@ void qSlicerAbstractModuleFactoryManager::registerModule(const QFileInfo& file)
     {
     if (d->Verbose)
       {
-      qDebug() << " checking file: " << file.absoluteFilePath() << " as a " << typeid(*factory).name();
+      qDebug() << " checking file: " << file.absoluteFilePath() << " as a " << typeidfactory->name();
```
Comment (jcfr) [inline Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx:302]: This change is likely a mistake

File: Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx [hunk 3]
```diff
@@ -299,15 +299,15 @@ void qSlicerAbstractModuleFactoryManager::registerModule(const QFileInfo& file)
...
       }
     if (!factory->isValidFile(file))
       {
       continue;
       }
     if (d->Verbose)
       {
-      qDebug() << " recognized file: " << file.absoluteFilePath() << " as a " << typeid(*factory).name();
+      qDebug() << " recognized file: " << file.absoluteFilePath() << " as a " << typeidfactory->name();
```
Comment (jcfr) [inline Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx:310]: This change is likely a mistake

File: Libs/MRML/Core/vtkDataFileFormatHelper.cxx [hunk 4]
```diff
@@ -160,7 +160,7 @@ void vtkDataFileFormatHelper::PopulateITKSupportedWriteFileTypes()
         // Once we have the GetSupportedWriteGenericNames() ready, we will used that.
         // For now, just use the class name.
         ITKImageFileFormat structFileFormat =
-          {io->GetNameOfClass(), io->GetNameOfClass(), io->GetNameOfClass(),(*writeItr).c_str()};
+          {io->GetNameOfClass(), io->GetNameOfClass(), io->GetNameOfClass(),writeItr->c_str()};
```
Comment (jcfr) [inline Libs/MRML/Core/vtkDataFileFormatHelper.cxx:163]:
```suggestion
          {io->GetNameOfClass(), io->GetNameOfClass(), io->GetNameOfClass(), writeItr->c_str()};
```

File: Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx (hunk 2 above)
Comment (lassoan) [inline Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx:302]: is this correct?

File: Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx (hunk 3 above)
Comment (lassoan) [inline Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx:310]: is this correct?

File: Libs/MRML/Widgets/qMRMLSortFilterProxyModel.cxx (hunk 1 above)
Comment (lassoan) [inline Libs/MRML/Widgets/qMRMLSortFilterProxyModel.cxx:275]: is this correct?

**Review by lassoan — COMMENTED** [review]
Thank you! It looks good to me, too, except a few places - see comments inline.

**Review by jcfr — COMMENTED** [review]
Beside of a few incorrectly updated lines, looks good.

Last, I am also wondering it is is worth updating `Base/QTCLI/vtkSlicerCLIModuleLogic.cxx`, this is rather complex logic and ensuring we can understand its history (using blame) may take priority on this stylistic change.

cc: @lassoan

File: Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx (hunk 2 above)
Comment (hjmjohnson) [reply Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx]: bad oversight on my part.  Reverted.

Comment (hjmjohnson) [reply Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx]: Now fixed.

File: Libs/MRML/Core/vtkDataFileFormatHelper.cxx (hunk 4 above)
Comment (hjmjohnson) [reply Libs/MRML/Core/vtkDataFileFormatHelper.cxx]: extra space after comment in new forced push.

File: Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx (hunk 3 above)
Comment (hjmjohnson) [reply Base/QTCore/qSlicerAbstractModuleFactoryManager.cxx]: no. Fixed now in forced push.

**Comment by hjmjohnson** [comment]
FYI :  I reverted Base/QTCLI/vtkSlicerCLIModuleLogic.cxx so that the git blame is not affected.  The regex substitutions should work there in the future if the updates are desired in the future.

**Review by jcfr — APPROVED** [review]

**Pull request merged**